mono/metadata/object.c

```diff
@@ -3955,7 +3955,8 @@ mono_runtime_get_main_args (void)
 {
 	MONO_REQ_GC_UNSAFE_MODE;
 	MonoError error;
-	MonoArray *result = mono_runtime_get_main_args_checked (&error);
+	MonoArrayHandle *arg_array = mono_runtime_get_main_args_checked_handle (&error);
+	MonoArray *result = MONO_HANDLE_RAW (*arg_array);
 	mono_error_assert_ok (&error);
 	return result;
```

Comment:
Wouldn't that hand over a ephemeral MonoArray pointer to "outside of Mono"? I think this function is used by embedding wrappers and such, so we cannot change the signature